src/psyclone/dynamo0p3.py

```diff
@@ -8833,15 +8853,25 @@ def proxy_name_indexed(self):
         return self._name+"_proxy"
 
     @property
-    def name_indexed(self):
+    def proxy_declaration_name(self):
         '''
-        :returns: the name for this argument with an additional index \
-                  which accesses the first element for a vector argument.
+        :returns: the proxy name for this argument with the array \
+                  dimensions added if required.
         :rtype: str
         '''
         if self._vector_size > 1:
-            return self._name+"(1)"
-        return self._name
+            return self.proxy_name+"("+str(self._vector_size)+")"
+        return self.proxy_name
+
+    @property
+    def proxy_data_type(self):
```

Comment:
If proxy_data_type is a new method then it should be tested in dynamo0p3_test.py as a DynKernelArgument test. Apologies if I missed where this is.

Reply:
Again, I'll leave it up to you. If you think it is validly covered somewhere else (and you get 100% coverage so it must be being used somewhere) I'll let it go.

Reply:
OK, as with previous comment, please ignore. You have done this in tests in different files.